router/middleware.go

```diff
@@ -0,0 +1,95 @@
+package router
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"regexp"
+	"runtime/debug"
+	"strings"
+
+	"github.com/netlify/netlify-commons/tracing"
+)
+
+var bearerRegexp = regexp.MustCompile(`^(?:B|b)earer (\S+$)`)
+
+const versionHeaderTempl = "X-NF-%s-Version"
+
+type Middleware func(http.Handler) http.Handler
+
+func MiddlewareFunc(f func(w http.ResponseWriter, r *http.Request, next http.Handler)) Middleware {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			f(w, r, next)
+		})
+	}
+}
+
+func VersionHeader(serviceName, version string) Middleware {
+	return MiddlewareFunc(func(w http.ResponseWriter, r *http.Request, next http.Handler) {
+		next.ServeHTTP(w, r)
```

Comment:
Is this made in purporse?: We call the whole middleware chain and right after we add the version header?

Reply:
yeah that is my intent, we are setting the version header for the client, not our internal consumption. This makes it so that none of our code can override that header. 

Internal code can get access to the `Version` directly